src/node_api.cc

```diff
@@ -658,9 +658,16 @@ void napi_clear_last_error(napi_env env) {
 napi_status napi_set_last_error(napi_env env, napi_status error_code,
                                 uint32_t engine_error_code,
                                 void* engine_reserved) {
-  env->last_error.error_code = error_code;
-  env->last_error.engine_error_code = engine_error_code;
-  env->last_error.engine_reserved = engine_reserved;
+
+  if (env == nullptr) {
+    if (error_code == napi_ok) {
```

Comment:
Actually, those places were not accessing `env` in any way except as a whole, and passing it to functions/macros capable of dealing with a `nullptr` `env`, so it was actually OK.